pkg/pool-weighted/test/LiquidityBootstrappingPool.test.ts

```diff
@@ -1,2 +1,161 @@
+      await pool.setPublicSwap(owner, false);
+
+      expect(await pool.instance.swapEnabled()).to.be.false;
+
+      await expect(pool.swapGivenIn({ in: 1, out: 0, amount: fp(0.1) })).to.be.revertedWith('SWAPS_PAUSED');
+
+      await pool.setPublicSwap(owner, true);
+
+      expect(await pool.instance.swapEnabled()).to.be.true;
+    });
+
+    it('LP cannot pause swaps', async () => {
+      await expect(pool.setPublicSwap(lp, false)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
+  });
+
+  describe('joins', () => {
+    sharedBeforeEach('initialize pool', async () => {
+      await pool.init({ recipient: owner, initialBalances });
+
+      //const action = await actionId(pool.instance, '_onJoinPool');
+      //await pool.vault.grantRole(action, owner);
+    });
+
+    it('no public LPs', async () => {
+      await expect(pool.joinGivenIn({ recipient: lp, amountsIn: initialBalances })).to.be.revertedWith(
+        'SENDER_NOT_ALLOWED'
+      );
+    });
+
+    it('allows owner to join', async () => {
+      await expect(pool.joinGivenIn({ recipient: owner, from: owner, amountsIn: initialBalances })).to.not.be.reverted;
+    });
+  });
+
+  describe('update weights', () => {
+    sharedBeforeEach('deploy tokens', async () => {
+      const action = await actionId(pool.instance, 'updateWeightsGradually');
+      await pool.vault.grantRole(action, owner);
+    });
+
+    it('only allows owner to update weights', async () => {
+      const now = await currentTimestamp();
+
+      await expect(pool.updateWeightsGradually(lp, now, now, weights)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
+
+    context('call update weights', () => {
+      // startWeights = [fp(0.3), fp(0.55), fp(0.1), fp(0.5)];
+      const endWeights = [fp(0.15), fp(0.25), fp(0.55), fp(0.05)];
+      const halfWeights = [fp(0.225), fp(0.4), fp(0.325), fp(0.05)];
+      let now, startTime: BigNumber, endTime: BigNumber;
+
+      sharedBeforeEach('updateWeightsGradually', async () => {
+        now = await currentTimestamp();
+        startTime = now.add(MINUTE * 10);
+        endTime = startTime.add(MINUTE * 60);
+
+        await pool.updateWeightsGradually(owner, startTime, endTime, endWeights);
+      });
+
+      it('stores the params', async () => {
+        const updateParams = await pool.getGradualUpdateParams();
+
+        expect(updateParams.startTime).to.equalWithError(startTime, 0.001);
+        expect(updateParams.endTime).to.equalWithError(endTime, 0.001);
+        expect(updateParams.endWeights).to.equalWithError(endWeights, 0.001);
+      });
+
+      it('gets start weights if called early', async () => {
+        const normalizedWeights = await pool.getNormalizedWeights();
+
+        // Need to decrease precision
+        expect(normalizedWeights).to.equalWithError(pool.normalizedWeights, 0.0001);
+      });
+
+      it('gets end weights if called late', async () => {
+        await advanceTime(MINUTE * 100);
```

Comment:
It'd be good to have this be tied to the update duration more explicitly instead of it being a magic value